src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -55,13 +55,16 @@ export function DiscoverMainRoute(props: Props) {
     dataViewEditor,
   } = services;
   const [error, setError] = useState<Error>();
+  const [loading, setLoading] = useState(true);
   const [savedSearch, setSavedSearch] = useState<SavedSearch>();
-  const dataView = savedSearch?.searchSource?.getField('index');
   const [dataViewList, setDataViewList] = useState<DataViewListItem[]>([]);
   const [hasESData, setHasESData] = useState(false);
   const [hasUserDataView, setHasUserDataView] = useState(false);
   const [showNoDataPage, setShowNoDataPage] = useState<boolean>(false);
   const { id } = useParams<DiscoverLandingParams>();
+  useEffect(() => {
+    setLoading(true);
+  }, [id]);
 
```

Comment:
One thing that's worth to mention: In Discover main we have now a dedicated loading state. and the difference to before is, when you switch a saved search like
`new` => 'id1'
`id2` => 'id3'
`id3` => `new`

So we're starting in this cases with a clean state, rather than having to do a state `cleanup`. I think this is a cleaner approach for these cases. it also cleans up local state like queries that were not submitted, which led to a functional error initially when doing this change.